Library/Homebrew/cmd/pull.rb

```diff
@@ -7,51 +25,16 @@
 require "core_formula_repository"
 
 module Homebrew
-  def pull_url(url)
-    # GitHub provides commits/pull-requests raw patches using this URL.
-    url += ".patch"
-
-    patchpath = HOMEBREW_CACHE + File.basename(url)
-    curl url, "-o", patchpath
-
-    ohai "Applying patch"
-    patch_args = []
-    # Normally we don't want whitespace errors, but squashing them can break
-    # patches so an option is provided to skip this step.
-    if ARGV.include?("--ignore-whitespace") || ARGV.include?("--clean")
-      patch_args << "--whitespace=nowarn"
-    else
-      patch_args << "--whitespace=fix"
-    end
-
-    # Fall back to three-way merge if patch does not apply cleanly
-    patch_args << "-3"
-    patch_args << patchpath
-
-    begin
-      safe_system "git", "am", *patch_args
-    rescue ErrorDuringExecution
-      if ARGV.include? "--resolve"
-        odie "Patch failed to apply: try to resolve it."
-      else
-        system "git", "am", "--abort"
-        odie "Patch failed to apply: aborted."
-      end
-    ensure
-      patchpath.unlink
-    end
-  end
-
   def pull
-    if ARGV.empty?
-      odie "This command requires at least one argument containing a URL or pull request number"
-    end
-
     if ARGV[0] == "--rebase"
       odie "You meant `git pull --rebase`."
     end
+    if ARGV.named.empty?
+      odie "This command requires at least one argument containing a URL or pull request number"
+    end
+    do_bump = ((ARGV.include? "--bump") && !(ARGV.include? "--clean"))
```

Comment:
I find it easier to read if the arguments are in parentheses instead, i.e.:

```ruby
do_bump = ARGV.include?("--bump") && !ARGV.include?("--clean")
```

(But since this is purely a stylistic nit, feel free to ignore this advice.)